### PR TITLE
LPS-198626 Fix eq filter with DATE_TIME field when DBType is ORACLE

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-198626

### CODE REVIEW

**Comment by p-albuquerque** [comment]
follow-up https://github.com/liferay-objects/liferay-portal/pull/2737

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 1]
```diff
@@ -412,6 +412,14 @@ private Column<?, Object> _getColumn(
...
 	}
 
+	private String _getDateTimePattern(String value) {
+		if (value.length() > 20) {
+			return "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+		}
+
+		return "yyyy-MM-dd'T'HH:mm:ss";
+	}
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:422]: @p-albuquerque please check if we can use this [method](https://github.com/liferay/liferay-portal/blob/1206124b6f7b7d569ceaf75a45f8dffac6ab062f/modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java#L175). Thanks!

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by guilhermedcamacho** [comment]
Hey @p-albuquerque , as we discussed in our daily meeting yesterday, can you improve our test to cover this scenario? I think it's gonna be around [here](https://github.com/liferay/liferay-portal/blob/1206124b6f7b7d569ceaf75a45f8dffac6ab062f/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L1702).
This way we can run the suite `ci:test:object-environments` and check the behavior in other databases too.
cc. @carolmariaabb

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:422]: It works!

**Comment by p-albuquerque** [comment]
@guilhermedcamacho actually, I notice that this case is already covered, the [buildRange](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-rest-test-util/src/main/java/com/liferay/object/rest/test/util/BaseObjectEntryManagerImplTestCase.java#L100-L109) method it use "ge" and "le" also asserting with [the border limit](https://github.com/liferay/liferay-portal/blob/1206124b6f7b7d569ceaf75a45f8dffac6ab062f/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L1702-L1709) (which is the case that I'm fixing with this PR)

**Comment by guilhermedcamacho** [comment]
> @guilhermedcamacho actually, I notice that this case is already covered, the [buildRange](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-rest-test-util/src/main/java/com/liferay/object/rest/test/util/BaseObjectEntryManagerImplTestCase.java#L100-L109) method it use "ge" and "le" also asserting with [the border limit](https://github.com/liferay/liferay-portal/blob/1206124b6f7b7d569ceaf75a45f8dffac6ab062f/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L1702-L1709) (which is the case that I'm fixing with this PR)

So let's run the suite `ci:test:object-environments` in your PR, this way we can check if this behavior also happens in other databases too.
cc. @andreefarias

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:object-environments

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:object-environments

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward:force

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Pull request closed without merging**